### DISCUSSION TITLE
Allowing empty routes

### DISCUSSION THREAD

**Discussion by Bluenix2** [encode/starlette#1809] [Potential Issue]
## Summary

I have an API structured and use `Mount('/resource/{resource_id'})` however this causes issues, because now it is impossible for me to define a route for `/resource/{resource_id}`. I am forced to use `/resource/{resource_id}/`.

## Code examples

My code currently looks like this:

```python
Mount('/resources/{resource_id:int}', routes=[
    Route('/', ConfigEndpoint),
])
```

...however I would like to be able to define a route with no path like this:

```python
Mount('/resources/{resource_id:int}', routes=[
    Route('', ConfigEndpoint),
])
```

**Comment by adriangb** [comment]
I've run into this as well! I'm trying to think what the cons would be to removing [this check](https://github.com/encode/starlette/blob/31164e346b9bd1ce17d968e1301c3bb2c23bb418/starlette/routing.py#L208). I think we'd need to modify `Mount` in possibly backwards incompatible ways, but I'll have to check to make sure.
